x/zoneconcierge/keeper/ibc_packet_btc_headers.go

```diff
@@ -28,17 +30,26 @@ func (k Keeper) BroadcastBTCHeaders(ctx context.Context) error {
 	// TODO: Improve reorg handling efficiency - instead of sending from Consumer base to tip,
 	// we should send a dedicated reorg event and then send headers from the reorged point to tip
 
+	// Build a map for O(1) channel lookups
+	consumerChannelMap, err := k.buildConsumerChannelMap(ctx, consumerChannels)
```

Comment:
Can we construct this consumer map only once in the `EndBlocker` and pass as parameter for the funcs `BroadcastBTCHeaders` and `BroadcastBTCStakingConsumerEvents`